frontend/src/advising/ManageStudents/AssignTemplateToUserModal.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FunctionComponent } from "react";
+import { Checkbox } from "@material-ui/core";
+import styled from "styled-components";
+import { IFolderData, ITemplatePlan } from "../../models/types";
+import { getTemplates, TemplatesAPI } from "../../services/TemplateService";
+import { useSelector } from "react-redux";
+import { AppState } from "../../state/reducers/state";
+import { getAdvisorUserIdFromState } from "../../state";
+import CheckBoxIcon from "@material-ui/icons/CheckBox";
+import {
+  AssignTemplateModalProps,
+  GenericSearchAssignModal,
+  OptionsProps,
+} from "../../components/common/GenericSearchModal";
+
+const FolderName = styled.div`
+  font-weight: bold;
+  margin-left: 10px;
+  margin-top: 5px;
+`;
+
+const FolderTemplateListContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  margin-left: 15px;
+`;
+
+export const AssignTemplateToUserModal: FunctionComponent<AssignTemplateModalProps<
+  ITemplatePlan
+>> = props => {
+  const { userId } = useSelector((state: AppState) => ({
+    userId: getAdvisorUserIdFromState(state),
+  }));
+  const fetchTemplates = (
+    currentList: IFolderData[],
+    searchQuery: string,
+    page: number,
+    setIsLoading: (loading: boolean) => void,
+    setResponse: (response: any, newList: IFolderData[]) => void
+  ) => {
+    setIsLoading(true);
+    getTemplates(userId, searchQuery, page)
+      .then((response: TemplatesAPI) => {
+        setResponse(response, currentList.concat(response.templates));
+        setIsLoading(false);
+      })
+      .catch((err: any) => console.log(err));
```

Comment:
Should we do anything here for error handling beyond logging to console?

Reply:
Hmm yeah probably. I think overall we need a more generic way to handle errors because right now it's pretty mismatched. I'll talk to Alex about it and see if she can come up with a nice design/plan for it